frontend/app/controllers/Giraffe.scala

```diff
@@ -62,14 +67,27 @@ object Giraffe extends Controller {
     )
   }
 
+  def contributeRedirect = NoCacheAction { implicit request =>
+    val countryGroup = request.getFastlyCountry.getOrElse(CountryGroup.RestOfTheWorld)
+
+    val baseUrl: Uri = redirectToGiraffe(countryGroup).absoluteURL
+    val paramsToPropagate = request.queryString.map { case (k,v) => k }.toSeq
+    val urlWithParams = paramsToPropagate.foldLeft[Uri](baseUrl) { (url, param) =>
+      url ? (param -> request.getQueryString(param))
+    }
+
+    Redirect(urlWithParams, SEE_OTHER)
+  }
 
   def contributeUK = contribute(CountryGroup.UK)
   def contributeUSA = contribute(CountryGroup.US)
   def contributeAustralia = contribute(CountryGroup.Australia)
+  def contributeEurope = contribute(CountryGroup.Europe)
 
-  def thanksUK = thanks(CountryGroup.UK, routes.Giraffe.contributeUK().url)
-  def thanksUSA = thanks(CountryGroup.US, routes.Giraffe.contributeUSA().url)
-  def thanksAustralia = thanks(CountryGroup.Australia, routes.Giraffe.contributeAustralia().url)
+  def thanksUK = thanks(CountryGroup.UK, routes.Giraffe.contributeRedirect().url)
```

Comment:
routes.Giraffe.contributeRedirect().url could be a default param in thanks or simply called in the function rather than specified each time.